src/ssmDocument/commands/createDocumentFromTemplate.ts

```diff
@@ -86,14 +88,31 @@ export async function createSsmDocumentFromTemplate(): Promise<void> {
     }
 }
 
-async function openTextDocumentFromSelection(item: SsmDocumentTemplateQuickPickItem): Promise<vscode.TextDocument> {
-    const template: object = getDocumentTemplate(item.docType)
+function yamlToJson(yaml: string): string {
+    const jsonObject = YAML.parse(yaml)
+    return JSON.stringify(jsonObject, undefined, '\t')
+}
+
+async function openTextDocumentFromSelection(
+    item: SsmDocumentTemplateQuickPickItem,
+    extensionPath: string
+): Promise<vscode.TextDocument | undefined> {
+    const templateYamlContent = await readFileAsString(path.join(extensionPath, 'templates', item.filename))
+    const selectedDocumentFormat = await promptUserForDocumentFormat(['YAML', 'JSON'])
+
+    // user pressed escape and didn't select a format
+    if (selectedDocumentFormat === undefined) {
+        return
+    }
+
     let content: string
-    if (item.language === 'ssm-yaml') {
-        content = YAML.stringify(template)
+    let languageId: string
+    if (selectedDocumentFormat === 'YAML') {
+        content = templateYamlContent
+        languageId = 'ssm-yaml'
     } else {
-        content = JSON.stringify(template, undefined, '\t')
+        content = yamlToJson(templateYamlContent)
```

Comment:
I'm confused, why is `yamlToJson` called when `selectedDocumentFormat !== 'YAML`?

Reply:
By default the template content is YAML, so when the format is not Yaml, we convert to JSON. There are only two supported languages for Systems Manager documents: JSON and YAML.

Reply:
Oh, gotcha, I was under the impression that this meant that the document itself was in JSON, so why would you be converting from YAML to JSON.

Makes sense, maybe add a comment or document the file to clarify what's expected.

Reply:
Will do!